include/aie/Dialect/AIEVec/Transforms/Passes.td

```diff
@@ -37,60 +37,4 @@ def AIEVectorize : Pass<"aie-vectorize", "ModuleOp"> {
   ];
 }
 
-def LowerVectorToAIEVec : Pass<"lower-vector-to-aievec", "func::FuncOp"> {
-  let summary = "Convert standard Vector dialect operations to equivalent AIE "
-                "vector ops";
-  let dependentDialects = ["AffineDialect", // NOTE: Do we need this?
-                           "xilinx::aievec::AIEVecDialect",
-                           "arith::ArithDialect",
-                           "memref::MemRefDialect",
-                           "scf::SCFDialect",
-                           "vector::VectorDialect"];
-  let options = [
-    Option<"shiftParam", "shift", "unsigned", /*default=*/"0",
-      "Shift parameter for rounding and saturation">,
-    Option<"zeroOffset", "zero-offset", "unsigned", /*default=*/"0",
-     "Zero offset for indicating the location of zeroes in convolution filter "
-     "(useful for 16x16 scheme)">,
-    Option<"dupFactor", "dup-factor", "unsigned", /*default=*/"2",
-     "Duplication factor for each value in convolution filter "
-     "(useful for 8x8 scheme)">,
-    Option<"aieTarget", "aie-target", "std::string", /*default=*/"\"aie\"",
-     "Select AIE version: \\\"aie\\\" or \\\"aieml\\\". This will determine "
-     "the vector size and available operations.">,
-  ];
-}
-
-def CanonicalizeForAIEVec : Pass<"canonicalize-for-aievec", "func::FuncOp"> {
-  let summary = "Transform common vector ops into a subset of vector ops with "
-                "standard vector ops closer to AIEVec semantics.";
-  let dependentDialects = ["arith::ArithDialect",
-                           "vector::VectorDialect"];
-  let options = [
-    Option<"aieTarget", "aie-target", "std::string", /*default=*/"\"aie\"",
-     "Select AIE version: \\\"aie\\\" or \\\"aieml\\\". This will determine "
-     "the vector size and available operations.">,
-  ];
-}
-
-def AIEVecTransformation : Pass<"aievec-transformation", "func::FuncOp"> {
-  let summary = "Transform simple aievec ops into more complex aievec ops.";
-  let options = [
-    Option<"aieTarget", "aie-target", "std::string", /*default=*/"\"aie\"",
-     "Select AIE version: \\\"aie\\\" or \\\"aieml\\\". This will determine "
-     "the vector size and available operations.">,
-  ];
-}
-
-def AIEVecConvOpTransformation : Pass<"aievec-convop-transformation", "func::FuncOp"> {
-  let summary = "Transform simple aievec ops into aievec mul_conv or mac_conv ops.";
-  let options = [
-    Option<"shiftParam", "shift", "unsigned", /*default=*/"0",
-      "Shift parameter for rounding and saturation">,
-    Option<"aieTarget", "aie-target", "std::string", /*default=*/"\"aie\"",
-     "Select AIE version: \\\"aie\\\" or \\\"aieml\\\". This will determine "
-     "the vector size and available operations.">,
-  ];
-}
-
```

Comment:
I'm curious about the decision to use C++ instead of tablegen for the boilerplate. I usually see code move the other direction. Is it related to this comment?
```
// TODO: Create a common base class for all AIEVec pipeline options.

Reply:
Thanks for the context and explanation.